src/spring/azext_spring/gateway.py

```diff
@@ -207,26 +202,38 @@ def _validate_route_config_exist(client, resource_group, service, name):
         raise InvalidArgumentValueError("Route config " + name + " already exists")
 
 
-def _update_app_resource_id(client, resource_group, service, app_name, app_resource_id):
+def _create_or_update_gateway_route_configs(client, resource_group, service, name, route_properties,
+                                            app_name, routes_file, routes_json):
+    app_resource_id = _get_app_resource_id_by_name(client, resource_group, service, app_name)
+    route_properties = _create_or_update_routes_properties(routes_file, routes_json, route_properties)
+
+    if app_resource_id is not None:
+        route_properties.app_resource_id = app_resource_id
+    route_config_resource = models.GatewayRouteConfigResource(
+        properties=route_properties)
+    return client.gateway_route_configs.begin_create_or_update(resource_group, service, DEFAULT_NAME, name, route_config_resource)
+
+
+def _get_app_resource_id_by_name(client, resource_group, service, app_name):
     if app_name is not None:
         app_resource = client.apps.get(resource_group, service, app_name)
```

Comment:
NIT: this will result an additional network call, how about spelling the resource id in the client side?

Reply:
This is not only to construct the app resource Id, but also used to check if the app exist.

Reply:
This is not only to construct the app resource Id, but also used to check if the app exist.